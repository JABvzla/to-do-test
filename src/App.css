* {
  outline: none;
}

body {
  background-color: #282c34;
  padding-top: 30px;
}

.card {
  background-color: #fff;
  border-radius: 10px;
  padding: 10px;
  overflow: hidden;
  margin-bottom: 20px;
  width: 450px;
  margin-left: auto;
  margin-right: auto;
}
