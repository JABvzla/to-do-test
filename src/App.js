import React from "react";
import Add from "./componets/input-add";
import List from "./componets/list";
import "./App.css";
import useTodo from "./useToDo";

function App({add, todoList, toggle}) {
  return (
    <div className="App">
      <Add onAdd={add} />
      <List todoList={todoList} onChecked={toggle} />
    </div>
  );
}

const conectedApp = () => <App {...useTodo()} />

export default conectedApp;
