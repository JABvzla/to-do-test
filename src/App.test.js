import React from "react";
import { render, fireEvent } from "@testing-library/react";
import App from "./App";

test("should render correctly", () => {
  const { queryByText, queryByTestId, asFragment } = render(<App />);
  const emptyList = queryByText(/No hay nada por hacer/i);
  const inputAdd = queryByTestId("input-add");
  const buttonAdd = queryByTestId("button-add");

  expect(inputAdd).toBeInTheDocument();
  expect(buttonAdd).toBeInTheDocument();
  expect(emptyList).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test("should add an item", () => {
  const { queryByText, queryByTestId } = render(<App />);
  const emptyList = queryByText(/No hay nada por hacer/i);
  const inputAdd = queryByTestId("input-add");
  const buttonAdd = queryByTestId("button-add");

  fireEvent.change(inputAdd, { target: { value: "item nuevo 2" } });
  fireEvent.click(buttonAdd);

  const newItem = queryByText(/item nuevo/i);

  expect(inputAdd.value).toBe("");
  expect(newItem).toBeInTheDocument();
  expect(inputAdd).toBeInTheDocument();
  expect(buttonAdd).toBeInTheDocument();
  expect(emptyList).not.toBeInTheDocument();
});

test("dont should add empty an item", () => {
  const { queryByText, queryByTestId, asFragment } = render(<App />);
  const emptyList = queryByText(/No hay nada por hacer/i);
  const inputAdd = queryByTestId("input-add");
  const buttonAdd = queryByTestId("button-add");

  fireEvent.change(inputAdd, { target: { value: "" } });
  fireEvent.click(buttonAdd);

  expect(inputAdd).toBeInTheDocument();
  expect(buttonAdd).toBeInTheDocument();
  expect(emptyList).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
