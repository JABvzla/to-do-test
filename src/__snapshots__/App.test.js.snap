// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dont should add empty an item 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="card input-add-wrapper"
    >
      <input
        class="input-add"
        data-testid="input-add"
        type="text"
        value=""
      />
      <input
        class="button-add"
        data-testid="button-add"
        type="button"
        value="Agregar"
      />
    </div>
    <div
      style="color: rgb(170, 170, 170); text-align: center;"
    >
      No hay nada por hacer :'(
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="card input-add-wrapper"
    >
      <input
        class="input-add"
        data-testid="input-add"
        type="text"
        value=""
      />
      <input
        class="button-add"
        data-testid="button-add"
        type="button"
        value="Agregar"
      />
    </div>
    <div
      style="color: rgb(170, 170, 170); text-align: center;"
    >
      No hay nada por hacer :'(
    </div>
  </div>
</DocumentFragment>
`;
